### PR TITLE
chore: introduce decoding quota to http_request in SNS aggregator canister

### PR DESCRIPTION
**Pull request by mraszyk**
This PR introduces the decoding quota to the `http_request` method of the SNS aggregator canister. Setting a decoding quota of 10000 (corresponding to roughly 10KB of decoded data incl. overhead) prevents the SNS aggregator canister from wasting instructions on parsing large HTTP requests which are not expected to be used (the SNS aggregator supports GET requests for its assets and thus no HTTP requests should be larger than a few KB). I've tested the MR manually using dfx (successfully retrieving the candid interface and the landing page and failing to make a large HTTP request which is what this PR addresses).

### CODE REVIEW

**Review by bitdivine — APPROVED** [review]

**Comment by dskloetd** [comment]
@anchpop and @aterga would probably be the right people to review this.

File: CHANGELOG-Sns_Aggregator.md [hunk 1]
```diff
@@ -19,6 +19,8 @@ The SNS Aggregator is released through proposals in the Network Nervous System.
 ### Fixed
 ### Security
 
+- Decoding quota of 10,000 in the `http_request` method.
```
Comment (aterga) [inline CHANGELOG-Sns_Aggregator.md:22]:
```suggestion
- Decoding quota of 10,000 bytes in the `http_request` method.
```
?

File: rs/sns_aggregator/src/lib.rs [hunk 2]
```diff
@@ -130,16 +129,12 @@ fn tail_log(limit: Option<u16>) -> String {
 
 /// Web server
 #[candid_method(query)]
-#[export_name = "canister_query http_request"]
-fn http_request(/* req: HttpRequest */) /* -> HttpResponse */
-{
-    ic_cdk::setup();
-    let request = call::arg_data::<(HttpRequest,)>(ic_cdk::api::call::ArgDecoderConfig::default()).0;
-    let response = match request.url.as_ref() {
+#[ic_cdk_macros::query(decoding_quota = 10000)]
```
Comment (aterga) [inline rs/sns_aggregator/src/lib.rs:132]: Could you explain in a comment where does this quota come from, please?

**Review by aterga — CHANGES_REQUESTED** [review]
LGTM modulo testing — could we please add a test to demonstrate that the `http_request` function works expectedly with inputs that (1) meet quota and (2) exceed quota by 1?

I see from the description that you've already tested this manually, I think that's enough in this case. Approving

File: CHANGELOG-Sns_Aggregator.md (hunk 1 above)
Comment (mraszyk) [reply CHANGELOG-Sns_Aggregator.md:22]: The decoding quota follows its own [cost](https://github.com/dfinity/candid/blob/f324a1686d6f2bd4fba9307a37f8e3f90cc7222b/rust/candid/src/de.rs#L170) model, i.e., it is not expressed in bytes.

**Review by aterga — APPROVED** [review]

File: rs/sns_aggregator/src/lib.rs (hunk 2 above)
Comment (mraszyk) [reply rs/sns_aggregator/src/lib.rs:136]: Done: [60b7d1d](https://github.com/dfinity/nns-dapp/pull/5004/commits/60b7d1dd9b27e58001a5f1313c286e7464dd477e)

File: CHANGELOG-Sns_Aggregator.md (hunk 1 above)
Comment (aterga) [reply CHANGELOG-Sns_Aggregator.md:22]: Would it make sense in the doc to explain what the unit means?

Comment (aterga) [reply CHANGELOG-Sns_Aggregator.md:22]: I think the explanation you've added in the code is enough, thanks!

**Review by aterga — APPROVED** [review]

**Pull request merged**